torch/csrc/utils/python_arg_parser.cpp

```diff
@@ -479,8 +562,26 @@ static void extra_kwargs(FunctionSignature& signature, PyObject* kwargs, ssize_t
   throw TypeError("invalid keyword arguments");
 }
 
-bool FunctionSignature::parse(PyObject* args, PyObject* kwargs, PyObject* dst[],
-                              bool raise_exception) {
+/*
+ * Like list.insert(), but for C arrays of PyObject*. Skips error checking.
+ */
+static void
+pyobject_array_insert(PyObject **array, int length, int index, PyObject *item)
+{
+    int j;
+
+    for (j = length; j > index; j--) {
+        array[j] = array[j - 1];
+    }
+    array[index] = item;
+}
+
+auto FunctionSignature::parse(PyObject* args, PyObject* kwargs, PyObject* dst[],  // NOLINT
+                              PyObject* overloaded_args[], bool raise_exception) -> bool {  // NOLINT
+  int num_overloaded_args = 0;
+  int arg_index = 0;
+  int j;
+  int new_class = 1;
```

Comment:
nit: make this a boolean, and move it closer to its use site. A decent chunk of these locals don't need to be available for the entire scope. This ain't old-school C!